bentoml/service/artifacts/__init__.py

```diff
@@ -48,17 +56,37 @@ def name(self):
         """
         return self._name
 
-    def pack(self, model):
+    def pack(self, model, metadata=None):  # pylint: disable=unused-argument
         """
         Pack the in-memory trained model object to this BentoServiceArtifact
 
         Note: add "# pylint:disable=arguments-differ" to child class's pack method
         """
+        if metadata:
```

Comment:
I think it would be cleaner to move this and similarly the code in `load` below to the `__getattribute__ ` method in this class.  That way we can avoid modifying all the other artifact classes. 

Everything else looks great! Thanks for the PR @jackyzha0! Sorry about the delay in getting to help with the review.